src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -3620,73 +3668,53 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
 
     void visit_If(const ASR::If_t &x) {
         this->visit_expr_wrapper(x.m_test, true);
-        llvm::Value *cond=tmp;
-        llvm::Function *fn = builder->GetInsertBlock()->getParent();
-        llvm::BasicBlock *thenBB = llvm::BasicBlock::Create(context, "then", fn);
-        llvm::BasicBlock *elseBB = llvm::BasicBlock::Create(context, "else");
-        llvm::BasicBlock *mergeBB = llvm::BasicBlock::Create(context, "ifcont");
-        builder->CreateCondBr(cond, thenBB, elseBB);
-        builder->SetInsertPoint(thenBB);
-        for (size_t i=0; i<x.n_body; i++) {
-            this->visit_stmt(*x.m_body[i]);
-        }
-        builder->CreateBr(mergeBB);
-
-        start_new_block(elseBB);
-        for (size_t i=0; i<x.n_orelse; i++) {
-            this->visit_stmt(*x.m_orelse[i]);
-        }
-
-        start_new_block(mergeBB);
+        create_if_else(
+            tmp,
+            [this, x]() {
+                for (size_t i=0; i<x.n_body; i++) {
+                    this->visit_stmt(*x.m_body[i]);
+                }
+            },
+            [this, x]() {
+                for (size_t i=0; i<x.n_orelse; i++) {
+                    this->visit_stmt(*x.m_orelse[i]);
+                }
+            }
+        );
```

Comment:
Let's see how short one can make this:
```suggestion
        create_if_else(tmp,
            [=]() {
                for (size_t i=0; i<x.n_body; i++) {
                    this->visit_stmt(*x.m_body[i]);
                }
            }, [=]() {
                for (size_t i=0; i<x.n_orelse; i++) {
                    this->visit_stmt(*x.m_orelse[i]);
                }
            }
        );
```